src/postgresql/pgoltp.tcl

```diff
@@ -1595,14 +1610,21 @@ pg_result $result -clear
 return $lda
 }
 
-proc CreateUserDatabase { lda host port db tspace superuser superuser_password user password } {
+proc CreateUserDatabase { lda host port azure db tspace superuser superuser_password user password } {
```

Comment:
There is already #132 by @halilozanakgul that gave the functionality suggested in #122. This doesn't skip CreateuserDatabase completely but checks if the databases exists and is empty and if it is it doesn't do the user or database creation step. Would this be sufficient for this functionality or would it still be necessary to skip  CreateUserDatabase completely?

Reply:
There's a general problem that managed PostgreSQL services do not provide superuser access, so when using HammerDB you have no choice but to pass non-superuser details into the superuser fields. Usually, the initial database user generated by the managed service does have many privileges (e.g. creating or altering a database).

As long as the privileged user has permission to run all the commands in CreateUserDatabase, or at least  to run the commands that are used when the database/user already exist (after #132), HammerDB works. It is unclear whether this is true for all managed PostgreSQL services.

The problem with the Azure single server is two-fold:

1) In the connection string the username has to include @machine_name, which is used by an Azure load balancer to identify the database server. On the database server itself, the username does not include the machine name. The current implementation of HammerDB quite reasonably assumes the username used in the connection string to also be usable as the username inside the database for the purpose of CREATE ROLE and GRANT commands, but this is not the case. 

2) Even after this PR, which attempts to address the first point by stripping the @machine_name part from the username in CreateUserDatabase, I get the following error from the ALTER USER .. PASSWORD .. command in CreateUserDatabase.

```sql 
ALTER USER marco PASSWORD '...'
ERROR:  must be superuser to alter replication users
```

So it seems that Azure single server does not allow all of the steps in CreateUserDatabase. 

Perhaps this specific situation could be addressed, but overall it seems that an option to skip CreateUserDatabase would allow users to always avoid managed service idiosyncrasies, because the remainder of the benchmark only needs permission to create tables and functions.

Reply:
Marco, many thanks for the detailed update and explanation. So it looks clear that adding an option to skip CreateUserDatabase in the schema builds for both the TPROC-C and TPROC-H workloads (in both GUI and CLI) is the best option to solve this issue. This also keeps compatibility with non managed service implementations as it is then possible to test in a standard environment. 
This should also be distinct from any use of SSL for connections.